pkg/api/serialization_test.go

```diff
@@ -405,9 +406,15 @@ func TestSpecificKind(t *testing.T) {
 	}
 }
 
+// Keep this in sync with the respective upstream set
+var nonInternalRoundTrippableTypes = sets.NewString("List", "ListOptions", "PodExecOptions", "PodAttachOptions")
```

Comment:
It isn't obvious to me either, so I located the PRs which added those upstream:
PodAttachOptions: https://github.com/kubernetes/kubernetes/pull/12119
PodExecOptions: https://github.com/kubernetes/kubernetes/pull/6869
ListOptions: https://github.com/kubernetes/kubernetes/pull/5763
List: https://github.com/kubernetes/kubernetes/pull/2786

@smarterclayton may have a word here